libbeat/publisher/queue/wrapper.go

```diff
@@ -0,0 +1,93 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/publisher"
+)
+
+// Wraps Queue interface to add QoS functionality on top of it.
+type QueueWrapper struct {
```

Comment:
I don't think wrappers need to be exported. Keep number of exported symbols low if possible.